### PR TITLE
Tracks event for WooCommerce.com connect notice

### PR DESCRIPTION
**Pull request by hesahesa**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Add track events for connect to woocommerce.com notice in the woocommerce settings and woocommerce marketplace wp-admin page.

Closes # .

### How to test the changes in this Pull Request:

**WooCommerce settings and WooCommerce extensions page**

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow instruction here: https://github.com/woocommerce/woocommerce/pull/45536 to setup the notices
2. Verify that showing the notice, dismissing the notice, and clicking the hyperlink/button on the notice triggers track events (verify it by using the "Network" tab, filter it by "pixel" and check that the track event actions are fired)
image

#### List of tracks events (all prefixed by `wcadmin_` in tracks)
- woo_connect_notice_in_settings_shown
- woo_connect_notice_in_settings_clicked
- woo_connect_notice_in_settings_dismissed
- woo_connect_notice_in_marketplace_shown
- woo_connect_notice_in_marketplace_clicked
- woo_connect_notice_in_marketplace_dismissed

**Plugins list page**

1. Opt-in to allow tracking, set `woocommerce_allow_tracking` option to `yes`
2. Follow test instructions in https://github.com/woocommerce/woocommerce/pull/46082
3. Verify Tracks event is fired when notice is shown and connect link is clicked

#### List of tracks events (all prefixed by `wcadmin_` in tracks)

- woo_connect_notice_in_plugins_shown
- woo_connect_notice_in_plugins_clicked

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by gedex** [comment]
Tested and can confirm all events were sent. LGTM.

**Review by gedex — APPROVED** [review]

**Comment by manzurahammed** [comment]
Tested and found that all events sent the data. 

Screenshot 2024-05-02 at 8 08 21 PM

**Review by manzurahammed — APPROVED** [review]

**Comment by barryhughes** [comment]
Hi @gedex—you selected me for review but, though I have a broad understanding of Tracks, I'm not too familiar with other parts of the source tree you are changing. Were you mostly just seeking someone who can merge?

**Comment by gedex** [comment]
> Were you mostly just seeking someone who can merge?

This. Because we weren't able to merge, but now we're able to. I'll un-assign you.

File: plugins/woocommerce-admin/client/marketplace/components/notice/notice.tsx [hunk 1]
```diff
@@ -52,6 +54,14 @@ export default function Notice( props: NoticeProps ): JSX.Element | null {
...
 	};
 
+	useEffect( () => {
+		if ( isVisible && typeof onLoad === 'function' ) {
+			onLoad();
+		}
+		// only run once
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ isVisible ] );
+
```
Comment (gedex) [inline plugins/woocommerce-admin/client/marketplace/components/notice/notice.tsx:64]: cc @corsonr—as you introduced the component—for heads up of this change

**Comment by gedex** [comment]
Should we add Tracks event for the notice introduced by https://github.com/woocommerce/woocommerce/pull/46082, @mehamasum ?

It's for unconnected store, but opted-in store can send the event.

**Comment by gedex** [comment]
> It's for unconnected store, but opted-in store can send the event.

Added in https://github.com/woocommerce/woocommerce/pull/47003/commits/66660f2594c864848a74b3f8fb037f97c75b016c

**Review by mehamasum — APPROVED** [review]
Thanks for the latest update @gedex, worked as expected.

**Pull request merged**